### PR TITLE
Text search

### LINKED ISSUE #967: As a user I want to be able to quickly do a full-text search from a listing page
On Gazettes.Africa we can search from the country and year pages, which is useful and the search is automatically faceted to the correct country (and year, if set):

image

It would be useful to have this on the judgment listing pages (court detail pages), and most of the other listing pages.

On the court page, I suggest below the year navigation. Otherwise, just above the facets and listing.

The only pages we don't currently support this would be the taxonomy pages, because we can't facet on taxonomies yet.

### PR DESCRIPTION
**Pull request by actlikewill**
- Adds text search for listing pages
- Adds text search for gazettes

closes #967 

https://www.loom.com/share/72e6e54142ca474daf17028b9b802ab3?sid=13a3a7e8-cea1-4d14-ba19-cf0cacc65f0e

### CODE REVIEW

File: peachjam/templates/peachjam/_gazette_search.html
Comment (longhotsummer) [inline peachjam/templates/peachjam/_gazette_search.html:1]: Why do gazettes need their own special one? Can we unify it into the generic one?

**Review by longhotsummer — COMMENTED** [review]
Great. Just wondering about the need for the gazette-specific one

File: peachjam/templates/peachjam/_gazette_search.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<form action="{% url 'search:search' %}" method="get" class="d-flex mb-3">
+  <input type="text"
+         class="form-control flex-grow-1"
+         autofocus
+         name="q"
+         placeholder="Search {{ locality.name }} {{ year }} Gazettes"/>
+  <input type="submit" class="btn btn-outline-primary ms-2" value="Search"/>
+  {% if year %}<input type="hidden" name="year" value="{{ year }}"/>{% endif %}
+  {% if locality %}<input type="hidden" name="locality" value="{{ locality }}"/>{% endif %}
+  <input type="hidden" name="doc_type" value="Gazette"/>
```
Comment (actlikewill) [inline peachjam/templates/peachjam/_gazette_search.html:10]: @longhotsummer The gazette specific one facets on doc_type=Gazette. I could try adding doc_type facet to the generic one and adding the doc_type context to the listing pages.

Comment (actlikewill) [reply peachjam/templates/peachjam/_gazette_search.html:10]: @longhotsummer Let me add that

Comment (actlikewill) [reply peachjam/templates/peachjam/_gazette_search.html]: Changed :+1:

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**